### ISSUE TITLE
[BUG] Unexpected change in nuclei segmentation when adding second channel (Cellpose-SAM)

### ISSUE THREAD

**Issue by elsarosenblad** [MouseLand/cellpose#1407]
## Before you fill out this form:
**Did you review the FAQ?**
- [X] Yes, I reviewed the FAQ on https://cellpose.readthedocs.io/en/latest/faq.html

**Did you look through previous (open AND closed) issues posted on GH?**
- [x] Yes, I went through all issues I could see

### **Describe the bug**
I am observing different segmentation behavior between single-channel and dual-channel input when using Cellpose-SAM for grey scale images.

When running segmentation on a single channel:

- **DAPI only** → nuclei are segmented correctly, and masks closely follow nuclear boundaries.
- **Membrane only** → membranes are segmented correctly and closely follow membrane borders.  
  In these membrane-only images, some nuclei without membrane signal are present due to autofluorescence. These nuclei are also segmented correctly, with masks closely following their nuclear boundaries.

However, when running segmentation on a **dual-channel image (DAPI + membrane)**, the behavior changes:

- Nuclei with surrounding membrane staining are segmented approximately as full cells (expected).
- Nuclei without membrane signal are expanded beyond their nuclear boundaries (unexpected).

In other words, when both channels are present, nuclei lacking membrane staining are still segmented as larger objects rather than remaining constrained to nuclear boundaries.

The expectation was that:
- Membrane regions would be segmented where membrane signal is present.
- Nuclei without membrane signal would remain segmented at nuclear boundaries,
similar to the membrane-only case (and potentially even more robustly, since the DAPI channel contains strong nuclear signal rather than weaker autofluorescence).

There is no crash or error — segmentation completes successfully in both cases. The only difference is the change in object extent when the second channel is present.

I am trying to determine whether this is expected context-dependent behavior due to the channel-invariant training of Cellpose-SAM, greyscale mutli-channel or whether I may be missing a recommended preprocessing step for dual-channel input.

### **To Reproduce**
```
import numpy as np
from cellpose import models
from tifffile import imread

model = models.CellposeModel(gpu=True)

# SCENARIO 1: Single channel (membrane only) - works as expected
cy5_image = np.array(imread("chunk_cycle_Cy5.tif"))  # Shape: (H, W), dtype: uint16

labels, flows, styles = model.eval(cy5_image)
# Result: Nuclei are segmented correctly, masks closely follow nuclear boundaries


# SCENARIO 2: Dual channel (DAPI + membrane) - unexpected expansion of nuceli without membrane singal 
dapi_image = np.array(imread("chunk_cycle_DAPI.tif"))  # Shape: (H, W), dtype: uint16

# Stack channels on last axis: [DAPI, Cy5]
dual_channel_img = np.stack([dapi_image, cy5_image], axis=-1)  # Shape: (H, W, 2)

labels, flows, styles = model.eval(dual_channel_img)
# Result:
# - Nuclei WITH membrane signal → segmented as full cells (expected)
# - Nuclei WITHOUT membrane signal → expanded beyond nuclear boundary (unexpected)
```

### **Run log**
ENVIRONMENT:
- cellpose version: 4.0.8
- Python: 3.12.12
- torch version: 2.7.1+cu126
- GPU: CUDA enabled and working
- Image shape: (13638, 13333, 2) for dual channel

VERBOSE OUTPUT:
2026-02-19 10:50:44,257 [INFO] ** TORCH CUDA version installed and working. **
2026-02-19 10:50:44,257 [INFO] >>>> using GPU (CUDA)
2026-02-19 10:50:46,187 [INFO] >>>> loading model /home/elsa/.cellpose/models/cpsam
2026-02-19 10:50:47,253 [INFO] processing grayscale image with (13638, 13333) HW

### **Screenshots**
Three screenshots of a small region. First image shows DAPI segmentation masks on raw DAPI image background. Middle image shows the dual-channel segmentation on the raw DAPI image background. We can see that the masks are expanded. To the right is the raw membrane marker image to show there is no marker signal (but some autofluorescence from nuclei). 

<img width="1360" height="460" alt="Image" src="https://github.com/user-attachments/assets/017d8541-68d4-448b-87b5-d051d42ec7de" />
